src/components/exchange/AvailableNetworksv2.tsx

```diff
@@ -0,0 +1,233 @@
+    shouldUpdate: true,
+    type: ExchangeModalTypes.swap,
+  });
+  const convertAssetAndNavigate = useCallback(
+    (chosenNetwork: Network) => {
+      const newAsset = asset;
+
+      // we need to convert the mainnet asset to the selected network's
+      newAsset.mainnet_address =
+        networks?.[ethereumUtils.getChainIdFromNetwork(Network.mainnet)]
+          ?.address ?? asset.address;
+      newAsset.address =
+        networks?.[ethereumUtils.getChainIdFromNetwork(chosenNetwork)].address;
+      newAsset.uniqueId = `${asset.address}_${chosenNetwork}`;
+      newAsset.type = chosenNetwork;
+
+      goBack();
+      navigate(Routes.EXCHANGE_MODAL, {
+        params: {
+          fromDiscover: true,
+          ignoreInitialTypeCheck: true,
+          defaultOutputAsset: newAsset,
+          type: CurrencySelectionTypes.input,
+          showCoinIcon: true,
+          title: lang.t('swap.modal_types.get_symbol_with', {
+            symbol: newAsset?.symbol,
+          }),
+          onSelectCurrency: updateInputCurrency,
+        },
+        screen: Routes.CURRENCY_SELECT_SCREEN,
+      });
+    },
+    [asset, goBack, navigate, networks, updateInputCurrency]
+  );
+
+  const handlePressContextMenu = useCallback(
+    ({ nativeEvent: { actionKey: network } }) => {
+      const chosenNetwork = network;
+      convertAssetAndNavigate(chosenNetwork);
+    },
+    [convertAssetAndNavigate]
+  );
+
+  const handlePressButton = useCallback(() => {
+    const chosenNetwork = availableNetworks[0];
+    convertAssetAndNavigate(chosenNetwork);
+  }, [availableNetworks, convertAssetAndNavigate]);
+
+  const networkMenuItems = useMemo(() => {
+    return Object.values(networkInfo)
+      .filter(
+        ({ exchange_enabled, value }) =>
+          exchange_enabled &&
+          value !== Network.mainnet &&
+          !!networks[ethereumUtils.getChainIdFromNetwork(value)]
+      )
+      .map(netInfo => ({
+        actionKey: netInfo.value,
+        actionTitle: netInfo.name,
+        icon: {
+          iconType: 'ASSET',
+          iconValue: `${
+            netInfo.layer2 ? `${netInfo.value}BadgeNoShadow` : 'ethereumBadge'
+          }`,
+        },
+      }));
+  }, [networks]);
+
+  const MenuWrapper =
+    availableNetworks.length > 1 ? ContextMenuButton : React.Fragment;
+
+  return (
+    <>
+      <MenuWrapper
+        // @ts-expect-error overloaded props ContextMenuButton
+        menuConfig={{ menuItems: networkMenuItems, menuTitle: '' }}
+        isMenuPrimaryAction
+        onPressMenuItem={handlePressContextMenu}
+        useActionSheetFallback={false}
+      >
+        <ButtonPressAnimation
+          scaleTo={0.96}
+          onPress={availableNetworks.length === 1 ? handlePressButton : NOOP}
+        >
+          <Box
+            borderRadius={99}
+            padding="8px"
+            paddingHorizontal={{ custom: 16 }}
+            marginHorizontal={{ custom: marginHorizontal }}
```

Comment:
wondering where these paddings and margins are coming from

we need 19 for the button 
![image](https://user-images.githubusercontent.com/12115171/187804290-e5e7ba67-4ec3-4cc1-9fc6-0c037a1f3bde.png) maybe from line 140 instead

padding horizontal should be 8, which is already done in line 146
![image](https://user-images.githubusercontent.com/12115171/187804369-0a27f043-8c43-4ae3-83c0-0da2dfce2f11.png)

https://www.figma.com/file/SFfYzRoHlapEecS0LTLwhQ/Swap-Aggregator-V2?node-id=898%3A56903

Reply:
should be good to go, note because of the negative margins for the icons we need 8 + 4 to match spec

Reply:
ah right, cool